entity-framework/core/what-is-new/ef-core-8.0/whatsnew.md

```diff
@@ -1127,3 +1127,468 @@ In addition to the enhancements described above, EF8 Preview 2 also includes som
 
 - [Configuration to opt out of occasionally problematic SaveChanges optimizations](https://github.com/dotnet/efcore/issues/29916)
 - [Add convention types for triggers](https://github.com/dotnet/efcore/issues/28687)
+
+## New in EF8 Preview 4
+
+### Collections of primitive types
+
+A persistent question when using relational databases is what to do with collections of primitive types? That is, lists or arrays of integers, date/times, strings, and so on? If you're using PostgreSQL, then its easy to store these things using PostgreSQL's [built-in array type](https://www.postgresql.org/docs/current/arrays.html). For other databases, there are two common approaches:
+
+- Create a table with a column for the primitive type value and another column to act as a foreign key linking each value to its owner of the collection.
+- Serialize the primitive collection into some column type that is handled by the database--for example, serialize to and from a string.
+
+The first option has advantages in many situations--we'll take a quick look at it at the end of this section. However, it's not a natural representation of the data in the model, and if what you really have is a collection of a primitive type, then the second option can be more effective.
+
+Starting with Preview 4, EF8 now includes built-in support for the second option, using JSON as the serialization format. JSON works well for this since modern relational databases include built-in mechanisms for querying and manipulating JSON, such that the JSON column can, effectively, be treated as a table when needed, without the overhead of actually creating that table. These same mechanisms allow JSON to be passed in parameters and then used in similar way to table-valued parameters in queries--more about this later.
+
+> [!TIP]
+> The code shown here comes from [PrimitiveCollectionsSample.cs](https://github.com/dotnet/EntityFramework.Docs/tree/main/samples/core/Miscellaneous/NewInEFCore8/PrimitiveCollectionsSample.cs).
+
+#### Primitive collection properties
+
+EF Core can map any `IEnumerable<T>` property, where `T` is a primitive type, to a JSON column in the database. This is done by convention for public properties which have both a getter and a setter. For example, all properties  in the following entity type are mapped to JSON columns by convention:
```

Comment:
The public part with getter and setter doesn't seem specific to primitive collections, so maybe can be removed?

```suggestion
EF Core can map any `IEnumerable<T>` property, where `T` is a primitive type, to a JSON column in the database. This is done by convention for public properties which have both a getter and a setter. For example, all properties in the following entity type are mapped to JSON columns by convention:
```